src/libaktualizr/primary/initializer.cc

```diff
@@ -53,23 +48,67 @@ void Initializer::initEcuSerials() {
   if (primary_ecu_hardware_id.empty()) {
     primary_ecu_hardware_id = Utils::getHostname();
     if (primary_ecu_hardware_id == "") {
-      throw Error("Could not get current host name, please configure an hardware ID explicitely");
+      throw Error("Could not get current host name, please configure an hardware ID explicitly");
     }
   }
 
-  ecu_serials.emplace_back(Uptane::EcuSerial(primary_ecu_serial_local),
-                           Uptane::HardwareIdentifier(primary_ecu_hardware_id));
-
+  EcuSerials new_ecu_serials;
+  new_ecu_serials.emplace_back(Uptane::EcuSerial(primary_ecu_serial_local),
+                               Uptane::HardwareIdentifier(primary_ecu_hardware_id));
   for (const auto& s : secondaries_) {
-    ecu_serials.emplace_back(s.first, s.second->getHwId());
+    new_ecu_serials.emplace_back(s.first, s.second->getHwId());
   }
 
-  storage_->storeEcuSerials(ecu_serials);
-}
+  register_ecus = stored_ecu_serials.empty();
+  if (!stored_ecu_serials.empty()) {
+    // We should probably clear the misconfigured_ecus table once we have
+    // consent working.
+    std::vector<bool> found(stored_ecu_serials.size(), false);
+
+    EcuCompare primary_comp(new_ecu_serials[0]);
+    EcuSerials::const_iterator store_it;
+    store_it = std::find_if(stored_ecu_serials.cbegin(), stored_ecu_serials.cend(), primary_comp);
+    if (store_it == stored_ecu_serials.cend()) {
+      LOG_INFO << "Configured Primary ECU serial " << new_ecu_serials[0].first << " with hardware ID "
+               << new_ecu_serials[0].second << " not found in storage.";
+      register_ecus = true;
+    } else {
+      found[static_cast<size_t>(std::distance(stored_ecu_serials.cbegin(), store_it))] = true;
+    }
+
+    // Check all configured Secondaries to see if any are new.
+    for (auto it = secondaries_.cbegin(); it != secondaries_.cend(); ++it) {
+      EcuCompare secondary_comp(std::make_pair(it->second->getSerial(), it->second->getHwId()));
+      store_it = std::find_if(stored_ecu_serials.cbegin(), stored_ecu_serials.cend(), secondary_comp);
+      if (store_it == stored_ecu_serials.cend()) {
+        LOG_INFO << "Configured Secondary ECU serial " << it->second->getSerial() << " with hardware ID "
+                 << it->second->getHwId() << " not found in storage.";
+        register_ecus = true;
+      } else {
+        found[static_cast<size_t>(std::distance(stored_ecu_serials.cbegin(), store_it))] = true;
```

Comment:
```suggestion
        found[store_it - stored_ecu_serials.cbegin()] = true;
```

Reply:
Ah this code was just moved...

Reply:
Indeed, but I think I also wrote it originally. Can't remember why exactly I used `std::distance` here. Do you see a good reason to prefer one over the other?

Reply:
oh oops I thought I answered earlier, must have missed the button.

So after reading some docs, I can't really see the difference but the substraction is shorter and seems clearer to me.

Reply:
Fixed.